### PR TITLE
ipdx: auto update docs on version changes (support new release system)

### PR DESCRIPTION
**Pull request by laurentsenta**
Fully automate version changes

- provide a `{{< version >}}` shortcode that can be reused accross the repository
- store the current version (lotus, miner, and go) in a resource file
- load the lotus-cli and lotus-miner-cli markdown from github on build
- provide a github workflow to auto-update the versions
  - triggered every morning
  - triggered on workflow_dispatch

(similar to our work on [kubo](https://github.com/ipfs/ipfs-docs/pull/1081/files), but even better thanks to Hugo's templating features)

- [x] test the workflow
- [x] confirm the latest release from github is the "highest semantic version" -- this is dealt with in @galargh's [release PR](https://github.com/filecoin-project/lotus/pull/12096/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R209)

Contributes to https://github.com/filecoin-project/lotus/issues/12072

### CODE REVIEW

**Comment by laurentsenta** [comment]
Test PR:
https://github.com/laurentsenta/lotus-docs/pull/1

after running a few times:
https://github.com/laurentsenta/lotus-docs/actions/workflows/update-version.yml

File: .github/workflows/update-version.yml
Comment (laurentsenta) [reply .github/workflows/update-version.yml]: Looks like I don't! 🙏  fixed

File: content/en/storage-providers/operate/lotus-miner-cli.md
Comment (laurentsenta) [reply content/en/storage-providers/operate/lotus-miner-cli.md:4]: Thanks for the heads up, I updated the code to deal with both. I guess we'll have to wait for at least one release to try this out.

**Comment by laurentsenta** [comment]
Tested in https://github.com/laurentsenta/lotus-docs/pull/2
With the new versions pulled from https://github.com/galargh/lotus/releases

**Comment by laurentsenta** [comment]
@galargh we should be able to merge, the workflow will be working as soon as we complete a new release in the lotus repo.
Until then the workflow will fail on merge but not break the docs.

**Comment by BigLep** [comment]
@laurentsenta : There was a `miner/` tag (https://github.com/filecoin-project/lotus/releases/tag/miner%2Fv1.28.2) published 2024-08-15, which we believe should help unblock the PR.

**Comment by BigLep** [comment]
Per [2024-08-26 meeting with IPDX](https://www.notion.so/pl-strflt/FilOz-IPDX-co-c062ac03f68347dea32c11d1fc2c1315#82e5b65327ee49f99e3911ae557a33ea), it was agreed that we should manually create a github release for the `miner/v1.28.2` tag.  (This was started in https://github.com/filecoin-project/lotus/releases/edit/untagged-53cb53b937b9699b4d38).  This unblocks homebrew and this lotus-docs automation.

I believe a Lotus maintainer needs to do this.  @rjan90 : are you ok to take it (or we do it in screenshot together)?  I'm nervous to personally be doing GitHub releases through the UI without review.

**Comment by galargh** [comment]
> Per [2024-08-26 meeting with IPDX](https://www.notion.so/pl-strflt/FilOz-IPDX-co-c062ac03f68347dea32c11d1fc2c1315#82e5b65327ee49f99e3911ae557a33ea), it was agreed that we should manually create a github release for the `miner/v1.28.2` tag. (This was started in [filecoin-project/lotus/releases/edit/untagged-53cb53b937b9699b4d38](https://github.com/filecoin-project/lotus/releases/edit/untagged-53cb53b937b9699b4d38)). This unblocks homebrew and this lotus-docs automation.
> 
> I believe a Lotus maintainer needs to do this. @rjan90 : are you ok to take it (or we do it in screenshot together)? I'm nervous to personally be doing GitHub releases through the UI without review.

If you could publish it, that would be great and it'd unblock this PR. I think it'd be better if lotus maintainer was present when that happens in case there's some community communication that needs to happen. You can ping me and we can do it together if you prefer.

**Comment by rjan90** [comment]
Hey @galargh! I published the `miner/v1.28.2` release now: https://github.com/filecoin-project/lotus/releases/tag/miner%2Fv1.28.2.

Let me know if you need any additional things

**Comment by laurentsenta** [comment]
Team, thanks for the update,
I'm waiting for https://github.com/filecoin-project/github-mgmt/pull/76 to be able to update this branch and check the Fleek build succeeds.
Automation ran in https://github.com/laurentsenta/lotus-docs/pull/2
I'll update the PR with the commits in https://github.com/laurentsenta/lotus-docs/commits/main/

**Comment by laurentsenta** [comment]
@rjan90 @BigLep I've updated the PR, and we should be ready for review and merge.

Could you double-check, on the preview deploy, that `content/en/lotus/manage/lotus-cli.md` and `content/en/storage-providers/operate/lotus-miner-cli.md` are correctly loaded from Github on build.

If that's easier, please feel free to invite me to the fleek project I'll check this myself (firstname @ singulargarden.com)

**Review by rjan90 — APPROVED** [review]
The changes here looks good! I just need to double-check the preview deploy on Fleek before merging (encountered some issues when trying to log-in)

**Comment by rjan90** [comment]
Can confirm that I see the changes in the preview deploy on Fleek:

Screenshot 2024-09-25 at 17 06 17
Screenshot 2024-09-25 at 17 06 32
Screenshot 2024-09-25 at 17 08 57

**Pull request merged**
